### PR TITLE
[bugfix] PTP check for output folder.

### LINKED ISSUE #40486: QgsProcessingParameterFolderDestination default value issue in PointsToPath in the modeler
This is a regression spotted in 3.16.1 that was not present in 3.16.0.

When running PointstoPath in the modeler the output_text_dir parameter has a default value that is the temporary output directoy + "OUTPUT_TEXT_DIR", I have no clue why the name of the parameter is added to the path of the temporary directory but now this ensure that the algorithm always fails due to the fact that the directory does not exist.

See https://github.com/qgis/QGIS/blob/ae565f180ad5c8a83b86df2ac7cd85df95b4c96d/python/plugins/processing/algs/qgis/PointsToPaths.py#L113 where a value magically appears in the modeler.

Will provide a bugfix as a temporary measure for this algorithm,

### PR DESCRIPTION
**Pull request by roya0045**
## Description

Fixes #40486 

Unsure why an invalid path could be provided in the first place.

### CODE REVIEW

**Comment by roya0045** [comment]
For backport to 3.16.1 as it is broken. @nyalldawson do you have a clue why the behavior changed, and only in the modeler?

**Comment by uclaros** [comment]
I think it broke just to convince @nyalldawson to abandon this algorithm in favor of *Points to lines*! ;p

**Comment by roya0045** [comment]
@uclaros In the meantimes I still have models using this ;)

**Comment by nyalldawson** [comment]
> I think it broke just to convince @nyalldawson to abandon this algorithm in favor of Points to lines! ;p

I'm already convinced -- I'm just waiting for the PR to be made api compatible and address the review comments...

**Comment by nyalldawson** [comment]
@roya0045 can you check the ticket please -- i can't reproduce the issue

**Comment by roya0045** [comment]
@nyalldawson I tried just now with a test model on my setup and couldn't reproduce either. I have no clue why it behaved that way at work consistently. Though this validation step still feels like a good safety net.

**Comment by nyalldawson** [comment]
I don't like silently skipping the output if a non existent folder is specified by the user though - so we should really either make the folder automatically or raise an exception here. But neither of those approaches helps if the parameter is incorrectly specified though!

**Comment by roya0045** [comment]
@nyalldawson Went for the standard exception

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 __copyright__ = '(C) 2014, Alexander Bruy'
 
 import os
+from os.path import exists
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:25]:
```suggestion
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 2]
```diff
@@ -154,6 +155,9 @@ def processAlgorithm(self, parameters, context, feedback):
         if sink is None:
             raise QgsProcessingException(self.invalidSinkError(parameters, self.OUTPUT))
 
+        if text_dir and not(exists(text_dir)):
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:158]:
```suggestion
        if text_dir and not(os.path.exists(text_dir)):
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 3]
```diff
@@ -154,6 +155,9 @@ def processAlgorithm(self, parameters, context, feedback):
...
+            raise QgsProcessingException(self.tr("Provided directory for text output does not exists"))
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:159]:
```suggestion
            raise QgsProcessingException(self.tr("The text output directory does not exist"))
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**